### ISSUE TITLE
Discussion: BACKEND technologies

### ISSUE THREAD

**Issue by miggs125** [Seneca-CDOT/telescope#2]
Part of the [MVP features](https://github.com/Seneca-CDOT/telescope/blob/master/docs/overview.md#mvp-features) is to determine what technologies to use. This is a discussion on what technologies to use for the backend. We should be able to create a poll so everyone can voice their opinion.

**Comment by humphd** [comment]
Backend needs to be "node" with whatever set of deps people need.  At this stage (MVP), I'd avoid adding a lot of other services until we need them (e.g., db).

**Comment by cryolis** [comment]
Should we use some kind of framework like Express or node would be enough?

**Comment by jerryshueh** [comment]
I'd suggest that we get some sort of a baseline going before we start seeking higher level technologies. Like we discussed in class, a simple system that produces static HTML pages based on feed data is a good starting point. Looking through [npm](https://www.npmjs.com/), there are a number of existing technologies that can help us get started. 

For example, when I think of data manipulation in node, I immediately default to JSON. Some light digging lead me to [feedparser](https://www.npmjs.com/package/feedparser). We can use something like this to produce Javascript objects from the feed streams, and pull specific attributes from those objects to render a basic HTML page. Even plaintext would be fine... just get _something_ on the screen. Our "backend", at least for now, can just be some cached JSON files, and perhaps a text file with the list of feed URLs we want to grab content from. From here we can start thinking about configurations, such as minimal wait time, cache limits, or the feed properties we want to cache. After we get a feel for grabbing, parsing, and storing feed data in the node environment, we can discuss what sort of frameworks we can use to make it more robust.

What does everyone think? I'll have to do a bit more research on RSS/Atom parsers in node, but that's at least one bit we won't have to code from scratch - there are a bunch out there that already exist.

**Comment by manekenpix** [comment]
I'm with @jerryshueh, let's start with something simple, maybe using [http](https://nodejs.org/api/http.html), and [dotenv](https://www.npmjs.com/package/dotenv) for simple config files. Also, it'd be interesting to start adding labels so we can tag issues and PRs as mentioned [here](https://github.com/Seneca-CDOT/telescope/issues/4), and we could even use the wiki to add documentation (technologies/modules/libraries we want to use, versions...). I think right now we don't have access to the wiki an can't add labels, maybe @humphd can help us with that.

**Comment by Silvyre** [comment]
All for labels. We could also make use of [GitHub's project management functionalities](https://github.com/features/project-management/), such assigning [Milestones](https://help.github.com/articles/viewing-your-milestone-s-progress/) to MVP features (which can function to label/categorize Issues, as well).

**Comment by sukhbeersingh** [comment]
Since the discussions seems to be closing in at Node, maybe we can put in a basic server file. I can do that, if agreeable. Another thing to consider is which package manager we want to use. We're all familiar with npm, but from most of the projects I encountered in the last month, yarn is pretty popular too.

**Comment by manekenpix** [comment]
Before writing anything, I think it'd be a good idea to make a decision on these:
- npm vs yarn
- node 12.13 LTS vs node 13.01 current
- Linter (to avoid 60 different styles of code)

**Comment by UltimaBGD** [comment]
Jumping onto the labels discussion, it would be best to start that early, and I think that could be filed under another issue itself. While it may not be direct work on the project, keeping the list of issues easily indexible will be fairly crucial for an active team of 50+ people in terms of organization, and I think should be high priority before we really start going.

**Comment by humphd** [comment]
I think this is well in hand now in follow up issues/prs that address specifics of the above.  Closing in favour of what's happening there.
